src/windows/network_status.c

```diff
@@ -82,28 +84,40 @@ static close_callback _onClose = NULL;
 void window_network_status_open(const char* text, close_callback onClose)
 {
 	_onClose = onClose;
-	safe_strcpy(window_network_status_text, text, sizeof(window_network_status_text));
-
-	// Check if window is already open
-	rct_window *window = window_bring_to_front_by_class(WC_NETWORK_STATUS);
-	if (window != NULL)
-		return;
 
-	window = window_create_centred(420, 90, &window_network_status_events, WC_NETWORK_STATUS, WF_10 | WF_TRANSPARENT);
-
-	window->widgets = window_network_status_widgets;
-	window->enabled_widgets = 1 << WIDX_CLOSE;
-	window_init_scroll_widgets(window);
-	window->no_list_items = 0;
-	window->selected_list_item = -1;
-	window->frame_no = 0;
-	window->min_width = 320;
-	window->min_height = 90;
-	window->max_width = window->min_width;
-	window->max_height = window->min_height;
-
-	window->page = 0;
-	window->list_information_type = 0;
+	//Write FORMAT_BLACK to text
+	int no_lines = 0, font_height = 0;
+	char str[STATUS_MAX_BUFFER];
+	char *buffer = str;
+	str[0] = 0;
+	buffer = utf8_write_codepoint(buffer, FORMAT_BLACK);
+	safe_strcpy(buffer, text, STATUS_MAX_BUFFER - 5);
```

Comment:
Just use the existing string handling functions. There is no need to roll your own.